### DISCUSSION TITLE
How to move selected rectangle (with content) and merge layer (afterwards)?

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#510] [General]
Assume I loaded a *.jpg image, then selected a rectangle.
How can I move this rectangle (including content !!!!) to another position?
How can I merge the selection layer into underlying base image layer?

**Comment by hi5** [comment]
1. Just move the rectangle to where you want it (shortcut key <kbd>M</kbd>) - the rectangle should now be its own layer.
2. Assuming you keep this rectangle the active layer, Merge layers (down) by pressing <kbd>Ctrl</kbd>+<kbd>E</kbd>, or use the Layer menu, Merge down (or up) option.

**Comment by pstein** [comment]
@hi5:

Thank you, but your answer 1. is not complete. 
Meanwhile I found out that after I selected the rectangle I have to mark  the "Move and resize image layer" symbol in "layout" toolbox. 
Then and only then I can move the rectangle WITH  CONTENT.

Or is there a way to tell PD to automatically activate  "Move and resize image layer" after I select a rectangle?

Beside the auto-activate feature it would be great if for merge layer a similar auto-merge would exist:
As soon as the user draws a new rectangle or clicks somewhere outside the current rectangle an auto-merge should happen.

Is this possible or can this be added (at least optional via check box) in the next release?

**Comment by hi5** [comment]
That is what I described :)
I don't think that is / will be possible without pressing something because you are in the selection tool which you can move around or append/substract  other regions - if you want to move automatically after that it would be harder to modify the selection, I suspect that changing the selection is more important so that is why it is that way.
So just select, press <kbd>M</kbd>, now move, press <kbd>Ctrl</kbd>+<kbd>E</kbd>

You have one hand on your mouse, the other on the keyboard, it is incredibly fast.

But perhaps there will be a setting at some point that does what you want, but not at the moment as far as I know.

Edit: just so you know there are also two options to merge layers in the Image menu: 
merge visible layers and <kbd>Shift</kbd>+<kbd>Ctrl</kbd>+<kbd>E</kbd>
flatten image <kbd>Shift</kbd>+<kbd>Ctrl</kbd>+<kbd>F</kbd>
